Bottom: 458814da95aaea195973129e6b0ae35f599eba09
Top:    6e1d365e7e919adda7bf14bf91a0d7670e36945f
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-29 14:13:06 -0500

Refresh of TESTING--VirtualMachine

---
diff --git a/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/bacula.yml b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/bacula.yml
new file mode 100644
index 00000000..98016240
--- /dev/null
+++ b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/bacula.yml
@@ -0,0 +1,4 @@
+bacula_storage_device_name: "FileStorage"
+bacula_storage_device_type: File
+bacula_storage_device_archive_device: "/mnt/bacula-storage"
+bacula_storage_device_media_type: "disk0"
diff --git a/roles/bacula-restores/defaults/main.yml b/roles/bacula-restores/defaults/main.yml
index 22b89405..b72968f6 100644
--- a/roles/bacula-restores/defaults/main.yml
+++ b/roles/bacula-restores/defaults/main.yml
@@ -1,3 +1,3 @@
 ---
 
-bacula_storage: "/mnt/bacula-storage"
+bacula_storage_path: "/mnt/bacula-storage"
diff --git a/roles/bacula-restores/tasks/main.yml b/roles/bacula-restores/tasks/main.yml
index 6939a2fc..1e614c88 100644
--- a/roles/bacula-restores/tasks/main.yml
+++ b/roles/bacula-restores/tasks/main.yml
@@ -1,26 +1,11 @@
 ---
 # tasks file for bacula-restores
 
-- name: List bacula storage
-  find:
-    paths: "{{bacula_storage}}"
-    file_type: file
-    recurse: yes
-  register: bacula_restores_list
-
-- name: Create bacula storage symlinks
-  file:
-    path: '{{ item.path | replace("{{bacula_storage}}","{{bacula_dest}}") | dirname }}'
-    state: directory
-    force: yes
-    mode: 0777
-  with_items: '{{ bacula_restores_list.files }}'
-
-- name: Create bacula storage directories
-  file:
-    src: '{{ item.path }}'
-    dest: '{{ item.path | replace("{{bacula_storage}}","{{bacula_dest}}") }}'
-    state: link
-    force: yes
-    mode: '{{ item.mode }}'
-  with_items: '{{ bacula_restores_list.files }}'
+- name: Synchronize bacula storage to restore directory
+  synchronize:
+    dest: "{{bacula_dest}}/" # required. Path on the destination host that will be synchronized from the source; The path can be absolute or relative.
+    src: "{{bacula_storage_path}}/" # required. Path on the source host that will be synchronized to the destination; The path can be absolute or relative.
+    delete: yes # not required. Delete files in C(dest) that don't exist (after transfer, not before) in the C(src) path. This option requires C(recursive=yes).
+    archive: yes # not required. Mirrors the rsync archive flag, enables recursive, links, perms, times, owner, group flags and -D.
+    link_dest: "{{bacula_storage_path}}/" # not required. add a destination to hard link against during the rsync.
+  delegate_to: "{{ inventory_hostname }}"
